src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java

```diff
@@ -402,15 +419,142 @@ private StaticLangSettingsView generateSettingsClass(
+    stubInterface.releaseLevelAnnotation(
+        context
+            .getNamer()
+            .getReleaseAnnotation(packageMetadataConfig.releaseLevel(TargetLanguage.JAVA)));
+    stubInterface.name(name);
+    stubInterface.callableMethods(filterIncludeCallableMethods(methods));
+    stubInterface.hasLongRunningOperations(interfaceConfig.hasLongRunningOperations());
+
+    return stubInterface.build();
+  }
+
+  private StaticLangFileView<StaticLangGrpcStubView> generateGrpcStubClassFile(
+      GapicInterfaceContext context) {
+    StaticLangFileView.Builder<StaticLangGrpcStubView> fileView =
+        StaticLangFileView.<StaticLangGrpcStubView>newBuilder();
+
+    fileView.classView(generateGrpcStubClass(context));
+    fileView.templateFileName(GRPC_STUB_TEMPLATE_FILENAME);
+
+    String outputPath =
+        pathMapper.getOutputPath(context.getInterface(), context.getProductConfig());
+    String className = context.getNamer().getApiGrpcStubClassName(context.getInterfaceConfig());
+    fileView.outputPath(outputPath + "/stub/" + className + ".java");
+
+    // must be done as the last step to catch all imports
+    fileView.fileHeader(fileHeaderTransformer.generateFileHeader(context));
+
+    return fileView.build();
+  }
+
+  private StaticLangGrpcStubView generateGrpcStubClass(GapicInterfaceContext context) {
+    SurfaceNamer namer = context.getNamer();
+    GapicInterfaceConfig interfaceConfig = context.getInterfaceConfig();
+
+    addGrpcStubImports(context);
+
+    List<StaticLangApiMethodView> methods = generateApiMethods(context);
+
+    StaticLangGrpcStubView.Builder stubClass = StaticLangGrpcStubView.newBuilder();
+
+    stubClass.doc(serviceTransformer.generateServiceDoc(context, null));
+
+    String name = namer.getApiGrpcStubClassName(interfaceConfig);
+    stubClass.releaseLevelAnnotation(
+        namer.getReleaseAnnotation(packageMetadataConfig.releaseLevel(TargetLanguage.JAVA)));
+    stubClass.name(name);
+    stubClass.parentName(namer.getApiStubInterfaceName(interfaceConfig));
+    stubClass.settingsClassName(
+        getAndSaveNicknameForRootType(context, namer.getApiSettingsClassName(interfaceConfig)));
+    stubClass.directCallables(apiCallableTransformer.generateStaticLangDirectCallables(context));
+    stubClass.apiCallables(apiCallableTransformer.generateStaticLangApiCallables(context));
+    stubClass.callableMethods(filterIncludeCallableMethods(methods));
+    stubClass.hasDefaultInstance(interfaceConfig.hasDefaultInstance());
+    stubClass.hasLongRunningOperations(interfaceConfig.hasLongRunningOperations());
+
+    return stubClass.build();
+  }
+
+  private String getAndSaveNicknameForRootType(GapicInterfaceContext context, String nickname) {
+    SurfaceNamer namer = context.getNamer();
+    ModelTypeTable typeTable = context.getModelTypeTable();
+
+    String fqTypeName = namer.getRootPackageName() + "." + nickname;
```

Comment:
Hmm, maybe that is just my fault for not realizing that. If you think it's clear I am happy to leave it, otherwise maybe `fullTypeName` is a compromise?